xcube/webapi/ows/coverages/routes.py

```diff
@@ -0,0 +1,202 @@
+            )
+        elif content_type == 'text/html':
+            result = (
+                f'<html><title>Collection</title><body>'
+                f'<p>{collectionId}</p>'
+                f'</body></html>'
+            )
+        elif content_type == 'application/json':
+            result = get_coverage_as_json(ds_ctx, collectionId)
+        else:
+            result = get_coverage_data(
+                ds_ctx, collectionId, self.request.query, content_type
+            )
+        return await self.response.finish(result, content_type=content_type)
+
+
+# noinspection PyAbstractClass,PyMethodMayBeStatic
+@api.route('/catalog/collections/{collectionId}/coverage/domainset')
+class CoveragesDomainsetHandler(ApiHandler[CoveragesContext]):
+    """
+    Describe the domain set of a coverage
+
+    The domain set is the set of input parameters (e.g. geographical extent,
+    time span) for which the coverage is defined.
+    """
+    # noinspection PyPep8Naming
+    @api.operation(
+        operation_id='coveragesDomainSet',
+        summary='OGC API - Coverages - domain set',
+    )
+    async def get(self, collectionId: str):
+        domain_set = get_coverage_domainset(
+            self.ctx.datasets_ctx, collectionId
+        )
+        return self.response.finish(domain_set)
+
+
+# noinspection PyAbstractClass,PyMethodMayBeStatic
+@api.route('/catalog/collections/{collectionId}/coverage/rangetype')
+class CoveragesRangetypeHandler(ApiHandler[CoveragesContext]):
+    """
+    Describe the range type of a coverage
+
+    The range type describes the types of the data contained in the range
+    of this coverage. For a data cube, these would typically correspond to
+    the types of the variables or bands.
+    """
+    # noinspection PyPep8Naming
+    @api.operation(
+        operation_id='coveragesDomainSet',
+        summary='OGC API - Coverages - range type',
+    )
+    async def get(self, collectionId: str):
+        range_type = get_coverage_rangetype(
+            self.ctx.datasets_ctx, collectionId
+        )
+        return self.response.finish(range_type)
+
+
+@api.route('/catalog/collections/{collectionId}/coverage/metadata')
+class CoveragesMetadataHandler(ApiHandler[CoveragesContext]):
+    """
+    Return coverage metadata
+
+    The metadata is taken from the source dataset's attributes
+    """
+    # noinspection PyPep8Naming
+    @api.operation(
+        operation_id='coveragesMetadata',
+        summary='OGC API - Coverages - metadata',
+    )
+    async def get(self, collectionId: str):
+        return self.response.finish(
+            get_collection_metadata(self.ctx.datasets_ctx, collectionId)
+        )
+
+
+@api.route('/catalog/collections/{collectionId}/coverage/rangeset')
+class CoveragesRangesetHandler(ApiHandler[CoveragesContext]):
+    """
+    Handle rangeset endpoint with a "not supported" response
+
+    This endpoint has been deprecated, but we handle it to make clear that
```

Comment:
Where does it say that it is deprecated? My only reference is https://m-mohr.github.io/geodatacube-api/#tag/OGC-API-Coverages/operation/describe-coverage-rangeset and I didn't see any mention about it being deprecated there.

Reply:
The standard document also says nothing about deprecation: https://docs.ogc.org/DRAFTS/19-087.html#coverage-paths

Reply:
Yes, this is brand-new, red-hot news which I first heard of at this week's Testbed-19 meeting :). There is a PR. planned to be merged today, for the GDC API draft at <https://github.com/m-mohr/geodatacube-api/pull/1> containing this change. The corresponding change in the Coverages draft is scheduled for a SWG vote some time soon (and apparently no opposition is expected).

The GDC PR also removes the metadata endpoint, but since I'd already implemented it when I learned of the revisions, and since it's potentially useful to have, I thought I may as well leave it in.